k4MarlinWrapper/src/components/MarlinProcessorWrapper.cpp

```diff
@@ -209,14 +209,14 @@ StatusCode MarlinProcessorWrapper::execute() {
 
   if (sc.isFailure()) {
     // Register empty event
-    the_event = new lcio::LCEventImpl();
     debug() << "Registering empty Event for EDM4hep to LCIO conversion event in TES" << endmsg;
-    auto       pO     = std::make_unique<LCEventWrapper>(the_event, true);
+    auto       pO     = std::make_unique<LCEventWrapper>(std::make_unique<lcio::LCEventImpl>());
     StatusCode reg_sc = evtSvc()->registerObject("/Event/LCEvent", pO.release());
     if (reg_sc.isFailure()) {
       error() << "Failed to register empty LCIO Event" << endmsg;
       return reg_sc;
     }
+    the_event = static_cast<IMPL::LCEventImpl*>(static_cast<LCEventWrapper*>(pObject)->getEvent());
```

Comment:
This is causing the test to fail.

In this case here pObject is `nullptr` because we are in the `sc.isFailure()` still

You probably need to store the `LCEventImpl()` pointer already, unless you want to retrieve what was just stored.

Reply:
The idea was here to just get back the pointer to what I just stored. I figured, since the failure case returns before I get to here, I would always get a valid pointer here, but apparently that is not true?

Reply:
Ah wait... wrong object...